add docs for DANDI hub (#95)
dandi/handbook: docs/50_hub.md

# Using the DANDI Hub

> [DANDI Hub](http://hub.dandiarchive.org) is a JupyterHub instance in the cloud to interact with the data stored in DANDI, and is free to use for exploratory analysis of data on DANDI. Note that DANDI Hub is not intended for significant computation, but provides a place to introspect Dandisets and to perform some analysis and visualization of data.

Reviewer: just one sentence per line reformatting:

```suggestion
[DANDI Hub](http://hub.dandiarchive.org) is a JupyterHub instance in the cloud to interact with the data stored in DANDI, and is free to use for exploratory analysis of data on DANDI.
Note that DANDI Hub is not intended for significant computation, but provides a place to introspect Dandisets and to perform some analysis and visualization of data.
```

## Registration

To use the [DANDI Hub](http://hub.dandiarchive.org), you must first register for an account using the [DANDI website](http://dandiarchive.org).

## Choosing a server option

> When you start up the DANDI Hub, you will be asked to select across a number of server options. For basic exploration, Tiny or Base would most likely be appropriate. The DANDI Hub also currently offers Medium and Large options, which have more available memory and compute power. The "T4 GPU inference" server comes with an associated T4 GPU, and is intended to be used for applications that require GPU for inference. We request that users of this server be considerate of their usage of the DANDI Hub as a free community resource. Training large deep neural networks is not appropriate. A "Base (MATLAB)" server is also available, which provides a MATLAB cloud installation.

Reviewer: Reformat and adjust last sentence to mention the need for the license:

```suggestion
When you start up the DANDI Hub, you will be asked to select across a number of server options.
For basic exploration, Tiny or Base would most likely be appropriate.
The DANDI Hub also currently offers Medium and Large options, which have more available memory and compute power.
The "T4 GPU inference" server comes with an associated T4 GPU, and is intended to be used for applications that require GPU for inference.
We request that users of this server be considerate of their usage of the DANDI Hub as a free community resource.
Training large deep neural networks is not appropriate.
A "Base (MATLAB)" server is also available, which provides a MATLAB cloud installation but you would be required to provide your own license.
```

## Example notebooks

> The best way to share analyses on DANDI data is through the DANDI example notebooks. These notebooks are organized by `<DANDI id>/<group>/<analysis>/<notebook.ipynb>`. Dandiset contributors are encouraged to use these notebooks to demonstrate how to read, analyze, and visualize the data, and how to produce figures from associated scientific publications.

Reviewer: Split to one sentence per line and point to example-notebooks repo

```suggestion
The best way to share analyses on DANDI data is through the DANDI example notebooks.
These notebooks are maintained in https://github.com/dandi/example-notebooks repository which provides more information about their organization.
Dandiset contributors are encouraged to use these notebooks to demonstrate how to read, analyze, and visualize the data, and how to produce figures from associated scientific publications.
```

Author: @yarikoptic is there a reason you removed the sentence: "These notebooks are organized by `<DANDI id>/<group>/<analysis>/<notebook.ipynb>`."

Author: I geuss you are right, this is no longer needed since this information is now in the [example notebooks readme](https://github.com/dandi/example-notebooks)

# Using the DANDI Hub

> [DANDI Hub](http://hub.dandiarchive.org) is a JupyterHub instance in the cloud to interact with the data stored in DANDI, and is free to use for exploratory analysis of data on DANDI.

Reviewer: ```suggestion
[DANDI Hub](http://hub.dandiarchive.org) is a [JupyterHub](https://jupyterhub.readthedocs.io) instance in the cloud to interact with the data stored in DANDI, and is free to use for exploratory analysis of data on DANDI.
```

> Note that DANDI Hub is not intended for significant computation, but provides a place to introspect Dandisets and to perform some analysis and visualization of data.

Reviewer: I think here would be great to point to some tutorial/walkthrough or a video depicting general features of JupyterHub/Lab.  Later we could compliment with our own.  Do you know any good one?

Reviewer 2: Here is one [YouTube video](https://www.youtube.com/watch?v=5pf0_bpNbkw&t=09m20s) I found that has a good chapter on JupyterLab starting at 9m20s.

Reviewer: yeah -- seems to be nicely done, add a reference?

Reviewer 2: I have added a sentence in the suggestion below.

> [DANDI Hub](http://hub.dandiarchive.org) is a [JupyterHub](https://jupyterhub.readthedocs.io) instance in the cloud to interact with the data stored in DANDI, and is free to use for exploratory analysis of data on DANDI.
> Note that DANDI Hub is not intended for significant computation, but provides a place to introspect Dandisets and to perform some analysis and visualization of data.

Reviewer 2: ```suggestion
[DANDI Hub](http://hub.dandiarchive.org) is a [JupyterHub](https://jupyterhub.readthedocs.io) instance in the cloud to interact with the data stored in DANDI, and is free to use for exploratory analysis of data on DANDI.
For instructions on how to navigate JupyterHub see this [YouTube tutorial](https://www.youtube.com/watch?v=5pf0_bpNbkw&t=09m20s).
Note that DANDI Hub is not intended for significant computation, but provides a place to introspect Dandisets and to perform some analysis and visualization of data.
```

Author: @kabilar thanks for adding a link. This is a great video. It may be a bit strange to link to a random youtuber, but I can't find a similar video from the Jupyter team

## Registration

To use the [DANDI Hub](http://hub.dandiarchive.org), you must first register for an account using the [DANDI website](http://dandiarchive.org).

## Choosing a server option

When you start up the DANDI Hub, you will be asked to select across a number of server options.
For basic exploration, Tiny or Base would most likely be appropriate.
The DANDI Hub also currently offers Medium and Large options, which have more available memory and compute power.
The "T4 GPU inference" server comes with an associated T4 GPU, and is intended to be used for applications that require GPU for inference.
We request that users of this server be considerate of their usage of the DANDI Hub as a free community resource.
Training large deep neural networks is not appropriate.
A "Base (MATLAB)" server is also available, which provides a MATLAB cloud installation but you would be required to provide your own license.

## Example notebooks

> The best way to share analyses on DANDI data is through the DANDI example notebooks.

Reviewer: May be expand to

```suggestion
The best way to get started with existing data analyses or share your analyses on DANDI data is through the DANDI example notebooks.
```

Author: I would argue the best way to get started would be to start a notebook on the DANDI Hub, but I don't think users should use the "example notebooks" repo until they are ready to share them, so I prefer the text as it currently is.

> These notebooks are maintained in https://github.com/dandi/example-notebooks repository which provides more information about their organization.

Reviewer 2: ```suggestion
These notebooks are maintained in the [dandi/example-notebooks](https://github.com/dandi/example-notebooks) repository which provides more information about their organization.
```

> Dandiset contributors are encouraged to use these notebooks to demonstrate how to read, analyze, and visualize the data, and how to produce figures from associated scientific publications.

Reviewer: IMHO here it would be **very** valuable to show how to reference those from within dandiset metadata/meditor

Reviewer 2: Hi @yarikoptic, would it be best to reference these notebooks in the `Related resources` section of the Dandiset metadata?

Reviewer: here we are talking about hub... hub is particularly "good" for notebooks. I think it was a right call to mention them here (and may be elsewhere too).

Reviewer 2: Hi @yarikoptic, sorry, I am confused regarding this suggestion.

> Once this Pull Requests is accepted, your contributed notebook will be available to all DANDI Hub users.

Reviewer 2: ```suggestion
Notebooks can be added and updated through a pull request to the [dandi/example-notebooks](https://github.com/dandi/example-notebooks) repository.  Once the pull request is merged, your contributed notebook will be available to all DANDI Hub users.
```

Reviewer: I tuned it to retain 1 sentence per line as elsewhere in this PR.

```suggestion
Notebooks can be added and updated through a pull request to the [dandi/example-notebooks](https://github.com/dandi/example-notebooks) repository.
Once the pull request is merged, your contributed notebook will be available to all DANDI Hub users.
```

@kabilar FYI -- I love such a convention but it was not generally accepted: https://github.com/dandi/handbook/pull/99 but still might be worth voting there to reflect your opinion. As no other convention was formalized though, I consider it is as ok to have it 1 sentence per line here ;-)

Reviewer 2: Thanks @yarikoptic.  That sounds good.

> Notebooks can be added and updated through a pull request to the [dandi/example-notebooks](https://github.com/dandi/example-notebooks) repository.  Once the pull request is merged, your contributed notebook will be available to all DANDI Hub users.

Reviewer: just for consistency within this file on one sentence per line

```suggestion
Notebooks can be added and updated through a pull request to the [dandi/example-notebooks](https://github.com/dandi/example-notebooks) repository.
Once the pull request is merged, your contributed notebook will be available to all DANDI Hub users.
```

# Using the DANDI Hub

> The DANDI Hub is a JupyterHub instance in the cloud to interact with the data stored in DANDI, and is free to use for exploratory analysis of data on DANDI. Note that DANDI Hub is not intended for significant computation, but provides a place to introspect Dandisets and to perform some analysis and visualization of data.

Author: ```suggestion
[DANDI Hub](http://hub.dandiarchive.org) is a JupyterHub instance in the cloud to interact with the data stored in DANDI, and is free to use for exploratory analysis of data on DANDI. Note that DANDI Hub is not intended for significant computation, but provides a place to introspect Dandisets and to perform some analysis and visualization of data.
```
